driver-core/src/main/com/mongodb/internal/connection/CommandMessage.java

```diff
@@ -259,6 +263,15 @@ private BsonDocument getCommandToEncode() {
         return extraElements;
     }
 
+    private void checkServerVersionForTransactionSupport() {
+        int wireVersion = getSettings().getMaxWireVersion();
+        if (wireVersion < FOUR_DOT_ZERO_WIRE_VERSION
+                || (wireVersion < MAX_DRIVER_WIRE_VERSION && getSettings().getServerType() == SHARD_ROUTER)) {
```

Comment:
Replace MAX_DRIVER_WIRE_VERSION with FOUR_DOT_TWO_WIRE_VERSION.  Otherwise this will break the next time we have to increment MAX_DRIVER_WIRE_VERSION